client/struct.go

```diff
@@ -14,17 +14,24 @@ type GenericJSONResponse struct {
 	Data     interface{}
 }
 
-// SendResponse Warapper function to send Generic JSON response back to the rest client
-func SendResponse(w http.ResponseWriter, opRet int, opErrno int, opErrStr string, status int, v interface{}) {
-	var rsp GenericJSONResponse
+// FormResponse Warapper function to construct Generic JSON response to be sent
```

Comment:
Why did you split this into two? It was much simpler just calling SendResponse once, instead of making two calls.